roles/openshift_logging_fluentd/tasks/main.yaml

```diff
@@ -0,0 +1,258 @@
+    delete_after: true
+
+# create Fluentd secret
+- name: Retrieving the cert to use when generating secrets for Fluentd
+  slurp:
+    src: "{{ generated_certs_dir }}/{{ item.file }}"
+  register: key_pairs
+  with_items:
+    - { name: "ca_file", file: "ca.crt" }
+    - { name: "fluentd_key", file: "system.logging.fluentd.key"}
+    - { name: "fluentd_cert", file: "system.logging.fluentd.crt"}
+
+# TODO: add aggregation secrets if necessary
+- name: Set logging-fluentd secret
+  oc_secret:
+    state: present
+    kubeconfig: "{{ tempdir }}/admin.kubeconfig"
+    name: logging-fluentd
+    namespace: "{{ openshift_logging_namespace }}"
+    contents:
+    - path: ca
+      data: "{{ key_pairs | entry_from_named_pair('ca_file') | b64decode }}"
+    - path: key
+      data: "{{ key_pairs | entry_from_named_pair('fluentd_key') | b64decode }}"
+    - path: cert
+      data: "{{ key_pairs | entry_from_named_pair('fluentd_cert') | b64decode }}"
+
+# create Fluentd daemonset
+
+# this should change based on the type of fluentd deployment to be done...
+# TODO: pass in aggregation configurations
+- name: Generate logging-fluentd daemonset definition
+  template:
+    src: "{{ fluentd_version }}/fluentd.j2"
+    dest: "{{ tempdir }}/templates/logging-fluentd.yaml"
+  vars:
+    daemonset_name: logging-fluentd
+    daemonset_component: fluentd
+    daemonset_container_name: fluentd-elasticsearch
+    daemonset_serviceAccount: aggregated-logging-fluentd
+    app_host: "{{ openshift_logging_fluentd_app_host }}"
+    app_port: "{{ openshift_logging_fluentd_app_port }}"
+    ops_host: "{{ openshift_logging_fluentd_ops_host }}"
+    ops_port: "{{ openshift_logging_fluentd_ops_port }}"
+    fluentd_nodeselector_key: "{{ openshift_logging_fluentd_nodeselector.keys()[0] }}"
+    fluentd_nodeselector_value: "{{ openshift_logging_fluentd_nodeselector.values()[0] }}"
+  check_mode: no
+  changed_when: no
+
+- name: Set logging-fluentd daemonset
+  oc_obj:
+    state: present
+    kubeconfig: "{{ tempdir }}/admin.kubeconfig"
+    name: logging-fluentd
+    namespace: "{{ openshift_logging_namespace }}"
+    kind: daemonset
+    files:
+    - name: logging-fluentd
+      path: "{{ tempdir }}/logging-fluentd.yaml"
+    delete_after: true
+
+# Scale up Fluentd
+- name: Retrieve list of Fluentd hosts
+  oc_obj:
+    state: list
+    kind: node
+  when: "'--all' in openshift_logging_fluentd_hosts"
+  register: fluentd_hosts
+
+- name: Set openshift_logging_fluentd_hosts
+  set_fact:
+    openshift_logging_fluentd_hosts: "{{ fluentd_hosts.results.results[0]['items'] | map(attribute='metadata.name') | list }}"
+  when: "'--all' in openshift_logging_fluentd_hosts"
+
+- name: Label nodes for Fluentd deployment
+  oc_label:
+    name: "{{ fluentd_host }}"
+    kind: node
+    state: add
+    labels: "{{ openshift_logging_fluentd_nodeselector | oo_dict_to_list_of_dict }}"
+  with_items: "{{ openshift_logging_fluentd_hosts }}"
+  loop_control:
+    loop_var: fluentd_host
```

Comment:
Is there some way to have a delay in here?  For example, if you have 1000 nodes, it will clobber openshift to label all of those nodes at the same time.